presto-verifier/src/main/java/com/facebook/presto/verifier/rewrite/QueryRewriter.java

```diff
@@ -395,4 +430,271 @@ private static List<Property> applyPropertyOverride(List<Property> properties, L
                 .map(entry -> new Property(new Identifier(entry.getKey()), entry.getValue()))
                 .collect(toImmutableList());
     }
+
+    private static class NonDeterministicColumnRewriter
```

Comment:
Just asking: not sure about best java practice, but wouldn't it be better to have a new fairly large and separate class in a separate file?

Reply:
+1 on what @spershin has mentioned. And going beyond that, the right way is to
- add a `DefaultASTRewriter` that can traverse the entire tree with rewrite. This can also be used by `MaterializedViewOptimizationRewrite` and its `MaterializedViewQueryOptimizer`
- Inherit the `DefaultASTRewriter` here so you don't have to write a lot of empty visitors.

If we hard code the visitor here, we won't be able to handle all query shapes in the long term

Reply:
This is handled. First, moved this new ASTVisitor subclass out of QueryRewriter class and into a separate file. Second, separate the class into a general DefaultTreeRewriter and a concrete subclass FunctionCallRewriter of DefaultTreeRewriter.